src/components/calcite-label/calcite-label.tsx

```diff
@@ -0,0 +1,140 @@
+import {
+  Component,
+  Element,
+  Event,
+  EventEmitter,
+  Host,
+  h,
+  Prop,
+  Listen
+} from "@stencil/core";
+import { getElementTheme } from "../../utils/dom";
+
+@Component({
+  tag: "calcite-label",
+  styleUrl: "calcite-label.scss",
+  shadow: true
+})
+export class CalciteLabel {
+  //--------------------------------------------------------------------------
+  //
+  //  Element
+  //
+  //--------------------------------------------------------------------------
+
+  @Element() el: HTMLElement;
+
+  //--------------------------------------------------------------------------
+  //
+  //  Properties
+  //
+  //--------------------------------------------------------------------------
+
+  /** specify the status of the label and any child input / input messages */
+  @Prop({ mutable: true, reflect: true }) status: "invalid" | "valid" | "idle" =
+    "idle";
+
+  /** specify theme of the lavel and its any child input / input messages */
+  @Prop({ mutable: true, reflect: true }) theme: "light" | "dark" = "light";
+
+  /** is the wrapped element positioned inline with the label slotted text */
+  @Prop({ mutable: true, reflect: true }) appearance: "inline" | "inline-space-between" | "default" = "default";
+
+  //--------------------------------------------------------------------------
+  //
+  //  Lifecycle
+  //
+  //--------------------------------------------------------------------------
+
+  connectedCallback() {
+    let statusOptions = ["invalid", "valid", "idle"];
+    if (!statusOptions.includes(this.status)) this.status = "idle";
+
+    let theme = ["light", "dark"];
+    if (!theme.includes(this.theme)) this.theme = "light";
+  }
+
+  componentDidLoad() {
+    this.requestedInput = this.el.getAttribute("for");
+  }
+
+  render() {
+    const theme = getElementTheme(this.el);
+    const attributes = this.getAttributes();
```

Comment:
Yes, but it automatically spreads to the parent`calcite-component`- where we really want them is spread to the rendered `input` or `textarea` (or, in other components, some other element maybe).

So this passes any attributes directly to the rendered child element (and filters out props, which are just clutter on a non-Host element). This way users can add placeholder, autocomplete, validation patterns, etc. and they will be spread to the rendered child. 

Otherwise we'd need to explicitly define 35/40 possible input attributes (and still explicitly set them on the rendered child). Let me know if I'm missing something here.